tensorboard/components/tf_card_heading/util.js

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+/**
+ * Formats timestamp for the card header.
+ * @param {Date} date
+ * @return {string}
+ */
+export function formatDate(date) {
+  if (!date) {
+    return '';
+  }
+  // Turn things like "GMT-0700 (PDT)" into just "PDT".
+  return date.toString().replace(/GMT-\d+ \(([^)]+)\)/, '$1');
+}
+
+
+/**
+ * Returns CSS color that will contrast against background.
+ * @param {?string} background
+ * @return {string}
+ */
+export function pickTextColor(background) {
```

Comment:
This implementation of finding a contrasting color is really cool!

Reply:
I agree. So glad the w3c documented this.